### PR TITLE
Add linter + pre-commit hook

### PR DESCRIPTION
**Pull request by Simon-Busch**

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Ignore artifacts:
+.netlify
+_data
```
Comment (natevw) [inline .prettierignore:3]: I think also `public` (Gatsby output folder) should get added to this list too?

File: package.json [hunk 2]
```diff
@@ -106,12 +108,18 @@
   "scripts": {
     "build": "gatsby build",
     "develop": "gatsby develop",
-    "format": "prettier --write \"**/*.{js,jsx,ts,tsx,json,md}\"",
+    "format": "prettier --write \"*/**/*.{js,jsx,ts,tsx,json,css,scss}\"",
```
Comment (natevw) [inline package.json:111]: +1 to changing which extensions

but why change from `**/` to `*/**/` here? by adding a level it stops checking the top-level files which I don't think is intended?

File: package.json [hunk 3]
```diff
@@ -106,12 +108,18 @@
...
     "serve": "gatsby serve",
     "clean": "gatsby clean",
     "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1",
-    "validate": "node bin/validate.js"
+    "validate": "node bin/validate.js",
+    "prepare": "husky install"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "prettier --write \"*/**/*.{js,jsx,ts,tsx,json,css,scss}\""
```
Comment (natevw) [inline package.json:121]: same here as above, I think this should be:

```
"prettier --write \"**/*.{js,jsx,ts,tsx,json,css,scss}\""
```

i.e. without the extra `*/` at start of the pattern?

**Review by natevw — CHANGES_REQUESTED** [review]
Overall looks good, thanks! Just a couple tweaks I think to the ignore and to the file patterns it checks.

File: .prettierignore (hunk 1 above)
Comment (Simon-Busch) [reply .prettierignore:3]: Exactly, I'll add it right away

**Review by natevw — APPROVED** [review]

**Pull request merged**